netbsd/NetBSDProcess.c

```diff
@@ -0,0 +1,239 @@
+      .title = "CPU% ",
+      .description = "Percentage of the CPU time the process used in the last sampling",
+      .flags = 0,
+      .defaultSortDesc = true,
+   },
+   [PERCENT_NORM_CPU] = {
+      .name = "PERCENT_NORM_CPU",
+      .title = "NCPU%",
+      .description = "Normalized percentage of the CPU time the process used in the last sampling (normalized by cpu count)",
+      .flags = 0,
+      .defaultSortDesc = true,
+   },
+   [PERCENT_MEM] = {
+      .name = "PERCENT_MEM",
+      .title = "MEM% ",
+      .description = "Percentage of the memory the process is using, based on resident memory size",
+      .flags = 0,
+      .defaultSortDesc = true,
+   },
+   [USER] = {
+      .name = "USER",
+      .title = "USER      ",
+      .description = "Username of the process owner (or user ID if name cannot be determined)",
+      .flags = 0,
+   },
+   [TIME] = {
+      .name = "TIME",
+      .title = "  TIME+  ",
+      .description = "Total time the process has spent in user and system time",
+      .flags = 0,
+      .defaultSortDesc = true,
+   },
+   [NLWP] = {
+      .name = "NLWP",
+      .title = "NLWP ",
+      .description = "Number of threads in the process",
+      .flags = 0,
+   },
+   [TGID] = {
+      .name = "TGID",
+      .title = "TGID",
+      .description = "Thread group ID (i.e. process ID)",
+      .flags = 0,
+      .pidColumn = true,
+   },
+};
+
+Process* NetBSDProcess_new(const Settings* settings) {
+   NetBSDProcess* this = xCalloc(sizeof(NetBSDProcess), 1);
+   Object_setClass(this, Class(NetBSDProcess));
+   Process_init(&this->super, settings);
+   return &this->super;
+}
+
+void Process_delete(Object* cast) {
+   NetBSDProcess* this = (NetBSDProcess*) cast;
+   Process_done((Process*)cast);
+   free(this);
+}
+
+static void NetBSDProcess_writeField(const Process* this, RichString* str, ProcessField field) {
+   char buffer[256]; buffer[255] = '\0';
+   int attr = CRT_colors[DEFAULT_COLOR];
+
+   switch (field) {
+   // add NetBSD-specific fields here
+   default:
+      Process_writeField(this, str, field);
+      return;
+   }
+   RichString_appendWide(str, attr, buffer);
+}
+
+static int NetBSDProcess_compareByKey(const Process* v1, const Process* v2, ProcessField key) {
+   const NetBSDProcess* p1 = (const NetBSDProcess*)v1;
+   const NetBSDProcess* p2 = (const NetBSDProcess*)v2;
+
+   // remove if actually used
+   (void)p1; (void)p2;
+
+   switch (key) {
+   // add NetBSD-specific fields here
+   default:
+      return Process_compareByKey_Base(v1, v2, key);
+   }
+}
+
+const ProcessClass NetBSDProcess_class = {
+   .super = {
+      .extends = Class(Process),
+      .display = Process_display,
+      .delete = Process_delete,
+      .compare = Process_compare
+   },
+   .writeField = NetBSDProcess_writeField,
+   .compareByKey = NetBSDProcess_compareByKey
+};
+
+bool Process_isThread(const Process* this) {
+   return Process_isKernelThread(this) || Process_isUserlandThread(this);
+}
```

Comment:
Likely to be superseeded by #388 that changes thread handling to be performed globally. No longer used when #388 lands.

Reply:
Should be updated in 237e66c